### ISSUE TITLE
Feature request: start delay

### ISSUE THREAD

**Issue by melefabrizio** [pvolok/mprocs#95]
In my company we use mprocs to manage local development environments. In some cases a backend arrives at the point where it checks for a database connection before the database reached a steady state, so it would be nice to have a "dependency" system. Unfortunately that would be pretty difficult, as a process, as far as I can tell, is either running or stopped. A good workaround would be an `autostart_delay` property, to specify in ms how much time should pass between the `mprocs` command and the proc start.

**Comment by Thomas485** [comment]
Does

```
procs:
    my_backend:
        shell: sleep 0.001 && my_backend
        autostart: true
```

work for you as a workaround?

**Comment by melefabrizio** [comment]
Not in my particular case but it is not a bad workaround. Our processes are not commands but scripts relative to the cwd, so I have to put the sleep instruction inside the script, but it is not a problem for me.

Could be annoying if the process is started/stopped many times, the autostart delay I was proposing was specifically target to delay only the process autostart, and not subsequent starts.

**Comment by jayvdb** [comment]
A dependency system would be good.  This could be coupled with a mechanism for determining whether the service is `UP`, or something like a healthcheck.

**Comment by jayvdb** [comment]
Related, it would be helpful to have a remote control command to "sleep"

**Comment by pvolok** [comment]
> Related, it would be helpful to have a remote control command to "sleep"

What would the "sleep" command do?

**Comment by jayvdb** [comment]
sleep would take a number parameter , and sleep for that period of time.

**Comment by melefabrizio** [comment]
> What would the "sleep" command do?

More than a sleep it would behave more like a delay.
If a process has a `delay` option set in the configuration, when `mprocs` command is launched the process should wait for the set delay before starting, if it is set to `autostart`. When a process with a `delay` options is started manually, or restarted, it should not wait for the set delay (going with the assumption that when the whole configuration is already running it is stable, or the user himself takes care of the correct starting order of the processes).


A dependency system like the one mentioned by @jayvdb would be awesome for use cases like waiting for a database to be up before launching a backend, but I think is not so straight-forward detecting when a process is ready without the process itself signaling it.

**Comment by t-lock** [comment]
@melefabrizio I have a solution for the same/similar use case if you are interested.

In my case, I am running a backend, a database, and a hasura gql server in docker, and I want to open up my hasura console in the browser, as soon as the hasura service is ready. To achieve this, I've just passed a bash while loop that checks the availability of the hasura service, then calls my command:

```yaml
procs:
  backend:
    shell: "docker compose up"
    stop:
      # Ctrl C to stop (and mprocs will wait for it)
      send-keys: ["<C-c>"]
  hasura-console:
    # wait until the hasura service in docker is ready, then launch the console
    shell: 'while true; do response=$(curl -s -o /dev/null -w "%{http_code}" http://localhost:8080/v1/version); if [ $response -eq 200 ]; then cd hasura && hasura console; break; fi; sleep 1; done'
  client: "cd client && bun run dev -- --open"
```

**Comment by ankush** [comment]
same as https://github.com/pvolok/mprocs/issues/123
